packages/next/src/server/config.ts

```diff
@@ -236,6 +237,14 @@ function assignDefaults(
 
   const result = { ...defaultConfig, ...config }
 
+  // ensure correct default is set for api-resolver revalidate handling
+  if (!result.experimental?.trustHostHeader && ciEnvironment.hasNextSupport) {
+    if (!result.experimental) {
```

Comment:
This `result.experimental` check is duplicated in line 989, can we hoist it to ensure experimental is always set? like

```js
result = {
  ...defaultConfig, ...config,
  experimental: {
    ...defaultConfig.experimental,
    ...config?.experimental
  }
}
```

In this way we also always have the default and overriden values of all experimental flags